packages/config/src/log/messages.js

```diff
@@ -19,4 +21,24 @@ We recommend updating netlify.toml to set the functions directory under [functio
   )
 }
 
-module.exports = { warnLegacyFunctionsDirectory }
+const warnContextPluginConfig = function (logs, packageName, context) {
+  logWarning(
+    logs,
+    `"${packageName}" is installed in the UI, which means that it runs in all deploy contexts, regardless of file-based configuration.
+To run "${packageName}" in the ${context} context only, uninstall the plugin from the site plugins list.`,
+  )
+}
+
+const throwContextPluginsConfig = function (packageName, context) {
+  throwError(
+    `"${packageName}" is installed in the UI, which means that it runs in all deploy contexts, regardless of file-based configuration.
+To run "${packageName}" in the ${context} context only, uninstall the plugin from the site plugins list.
+To run "${packageName}" in all contexts, please remove the following section from "netlify.toml".
+
+  [[context.${context}.plugins]]
+  package = "${packageName}"
+`,
+  )
+}
```

Comment:
This looks alright to me, however I'm wondering if it would be worth just asking for a quick 👀 from someone from docs?

Reply:
Good idea :+1: 
cc @rstavchansky 

We can implement any copy change in a follow-up PR though, I think, because the text above was mostly taken from @verythorough's [comment here](https://github.com/netlify/pod-workflow/issues/192#issuecomment-824144116), and this feature is currently undocumented.